### DISCUSSION TITLE
Add example how to align tamagui theme with react-native-navigation into starter

### DISCUSSION THREAD

**Discussion by dohomi** [tamagui/tamagui#217] [General]
**Is your feature request related to a problem? Please describe.**
It would be good starting point to show how to align the styles of tamagui with the dark/light theme of `react-native-navigation`

**Describe the solution you'd like**
Currently the Expo dark theme will not work nicely out of the box with tamagui starter. The starter currently plays nicely with light mode but some colors seems off switching to dark. I will report back here if I find a working solution (https://reactnavigation.org/docs/themes#built-in-themes)

**Comment by thecaffeinatedengineer** [comment]
Hi @dohomi ,

I am having issues getting dark mode to work in the starter on native/expo. I stumbled on this whilst searching for related issues. So is this a known issue that it doesn't work yet ? If not I will raise an issue.

For reference this is what I get when I manually set the default theme to dark in the (expo) provider.

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-12-16 at 20 33 27](https://user-images.githubusercontent.com/14309350/208184391-14681023-38b6-408c-9be7-0a64b3e49fcc.png)

**Reply by thecaffeinatedengineer** [reply]
@natew not sure if you might know more about this ?

**Reply by ryanlntn** [reply]
Looks like react navigation let's you override the default and dark theme colors provided to native navigation so you can do something like: 

```tsx
// packages/app/provider/navigation/index.tsx
import { NavigationContainer, Theme } from '@react-navigation/native'
import * as Linking from 'expo-linking'
import { useMemo } from 'react'
import { useColorScheme } from 'react-native'
import config from '../../tamagui.config'

const themes: Record<'dark' | 'light', Theme> = {
  dark: {
    dark: true,
    colors: {
      background: config.themes.dark.background.val,
      border: config.themes.dark.borderColor.val,
      card: config.themes.dark.background.val,
      notification: config.themes.dark.background.val,
      primary: config.themes.dark.background.val,
      text: config.themes.dark.color.val,
    },
  },
  light: {
    dark: false,
    colors: {
      background: config.themes.light.background.val,
      border: config.themes.light.borderColor.val,
      card: config.themes.light.background.val,
      notification: config.themes.light.background.val,
      primary: config.themes.light.background.val,
      text: config.themes.light.color.val,
    },
  },
}

export function NavigationProvider({ children }: { children: React.ReactNode }) {
  const theme = useColorScheme()
  return (
    <NavigationContainer
      linking={useMemo(
        () => ({
          prefixes: [Linking.createURL('/')],
          config: {
            initialRouteName: 'home',
            screens: {
              home: '',
              'user-detail': 'user/:id',
            },
          },
        }),
        []
      )}
      theme={themes[theme ?? 'light']}
    >
      {children}
    </NavigationContainer>
  )
}
```

I also needed to update the Provider where it's mounted in the expo app to get Tamagui to switch the default theme as well:
```tsx
// apps/expo/App.tsx
<Provider defaultTheme={theme ?? 'light'}>
```

**Reply by thecaffeinatedengineer** [reply]
@ryanlntn thank for this.
